src/cmd/chattr.cpp

```diff
@@ -67,9 +67,23 @@ void Chattr::run(cxxopts::ParseResult& opts) {
             LOGD << "Setting attributes";
 
             if (attribute == "+public" || attribute == "+p") {
-                db->setPublic(true);
+
+                if (!db->isPublic()) {
+
+                    db->setPublic(true);
```

Comment:
Logic should probably be moved inside `setPublic`.

Reply:
Yes, it's cleaner 👍 https://github.com/DroneDB/DroneDB/pull/211/commits/7129d65985ed6b883584ef6d2c523581c12012d0